### PR TITLE
Always mark as read when archiving messages

### LINKED ISSUE #8309: Always mark read when archiving email messages
### Checklist

- [X] I have used the search function to see if someone else has already submitted the same feature request.
- [X] I will describe the problem with as much detail as possible.
- [X] This issue only contains a request for one single feature, **not** multiple (related) features.

### App version

6.804

### Problem you are trying to solve

I'd love some way to mark things as read when archiving.

Generally, when archiving something unread, it's because I have sufficient information from the subject line/already roughly know the contents - or know I don't care too much about it, but still want it "just in case". I like being able to filter by unread though, especially if it's something I want to get back to later and refile - but don't want in my inbox because it's not really a priority. 

### Suggested solution

An option to mark things read when archiving, like what's done with spam, would be great. Alternatively I think it'd be similarly nice to have one of the possible swipe actions be "mark as read and archive" together.

### Screenshots / Drawings / Technical details

_No response_

### PR DESCRIPTION
**Pull request by shamim-emon**
- Always mark read when archiving email messages (Behind feature flag).

- Fixed #8309

### CODE REVIEW

**Review by cketti — CHANGES_REQUESTED** [review]
Please check [this comment](https://github.com/thunderbird/thunderbird-android/issues/8309#issuecomment-2498120918) for additional requirements for this behavior change.

Unfortunately, this change doesn't achieve the desired effect.
1. `MessagingController.moveOrCopyMessageSynchronous()` is used for all move operations, not just for moving a message to the archive folder. However, we don't want all moved messages to automatically be marked as read.
2. Calling `setFlag()` on a `LocalMessage` instance will only mark the message as read locally, not on the server.

`MessagingController` always changes the local message state first, then enqueues a "command" that will perform the message changes on the server. By separating local changes from remote changes the app can be used offline. When connectivity returns the queue of remote operations will be processed.

There's already `MoveOrCopyFlavor.MOVE_AND_MARK_AS_READ`. Previously, there was no reason for `MessagingController.moveOrCopyMessageSynchronous()` to support this functionality. Now, adding support for `MOVE_AND_MARK_AS_READ` to this method would make it trivial to then change the behavior of only the archive action.

Please feel free to ask in the developer Matrix room if you have questions on any of this.

**Comment by shamim-emon** [comment]
> Please check [this comment](https://github.com/thunderbird/thunderbird-android/issues/8309#issuecomment-2498120918) for additional requirements for this behavior change.
> 
> Unfortunately, this change doesn't achieve the desired effect.
> 
> 1. `MessagingController.moveOrCopyMessageSynchronous()` is used for all move operations, not just for moving a message to the archive folder. However, we don't want all moved messages to automatically be marked as read.
> 2. Calling `setFlag()` on a `LocalMessage` instance will only mark the message as read locally, not on the server.
> 
> `MessagingController` always changes the local message state first, then enqueues a "command" that will perform the message changes on the server. By separating local changes from remote changes the app can be used offline. When connectivity returns the queue of remote operations will be processed.
> 
> There's already `MoveOrCopyFlavor.MOVE_AND_MARK_AS_READ`. Previously, there was no reason for `MessagingController.moveOrCopyMessageSynchronous()` to support this functionality. Now, adding support for `MOVE_AND_MARK_AS_READ` to this method would make it trivial to then change the behavior of only the archive action.
> 
> Please feel free to ask in the developer Matrix room if you have questions on any of this.

@cketti  please let me know if it's alright

**Comment by cketti** [comment]
I'm sorry. I missed the fact that the archive operation is treated as a move operation basically everywhere. That requires checking the feature flag in a lot of different places. It should just be one. I created #8623 so the feature flag will only have to be checked in one place.

**Comment by cketti** [comment]
With #8623 merged, please change your approach to modify the relevant code in `ArchiveOperations` instead.

**Comment by shamim-emon** [comment]
> With #8623 merged, please change your approach to modify the relevant code in `ArchiveOperations` instead.

@cketti Done! please let me know if anything else needed from my end.
Thanks

File: legacy/core/src/main/java/com/fsck/k9/controller/MessagingController.java
Comment (cketti) [inline legacy/core/src/main/java/com/fsck/k9/controller/MessagingController.java:1]: `moveOrCopyMessageSynchronous` still needs to be modified to support `MOVE_AND_MARK_AS_READ`. It looks like you accidentally dropped that change.

Please remove all unrelated formatting changes in this file.

File: legacy/core/src/main/java/com/fsck/k9/controller/ArchiveOperations.kt [hunk 1]
```diff
@@ -69,12 +73,21 @@ internal class ArchiveOperations(
...
     ) {
+        val operation = if (featureFlagProvider
+                .provide(
+                    FeatureFlagKey(key = "move_and_mark_as_read"),
+                ) == FeatureFlagResult.Enabled
+        ) {
+            MoveOrCopyFlavor.MOVE_AND_MARK_AS_READ
+        } else {
+            MoveOrCopyFlavor.MOVE
+        }
```
Comment (cketti) [inline legacy/core/src/main/java/com/fsck/k9/controller/ArchiveOperations.kt:84]:
```suggestion
        val operation = when (featureFlagProvider.provide("move_and_mark_as_read".toFeatureFlagKey())) {
            FeatureFlagResult.Enabled -> MoveOrCopyFlavor.MOVE_AND_MARK_AS_READ
            FeatureFlagResult.Disabled -> MoveOrCopyFlavor.MOVE
            FeatureFlagResult.Unavailable -> MoveOrCopyFlavor.MOVE
        }
```

**Review by cketti — CHANGES_REQUESTED** [review]

File: legacy/core/src/main/java/com/fsck/k9/controller/MessagingController.java
Comment (shamim-emon) [reply legacy/core/src/main/java/com/fsck/k9/controller/MessagingController.java:1]: @cketti Done!

**Review by cketti — APPROVED** [review]
Thanks for your pull request :+1:

Sorry for all the changes you needed to make to this branch. I added a couple of minor changes to avoid additional back and forth.

Right now this feature is disabled. The team will enable it for development and beta versions soon.

**Pull request merged**